test/Makefile.am

```diff
@@ -4,7 +4,8 @@ AM_TESTS_ENVIRONMENT = export OPENSSL_MODULES=${top_builddir}/src/.libs; \
 
 LDADD = @OPENSSL_LIBS@
 AM_LDFLAGS = @OPENSSL_LDFLAGS@
-AM_CFLAGS = -pthread
+AM_CFLAGS = -pthread -Wall -Werror
+AM_CPPFLAGS = @OPENSSL_INCLUDES@
```

Comment:
why didnt you put it with CFLAGS ?

Reply:
okay maybe one ones to include the headers fro  sources other than C in the future ..

Reply:
Because technically `-I` are preprocessor flags: if autotools selected a different compiler you could end up with separate preprocess, compile, assemble, link steps and it would result in errors.